modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,448 @@
+
+		/// Origin used by Oracles. Used to relay information from the Relaychain.
+		type OracleOrigin: EnsureOrigin<Self::Origin, Success = Vec<u8>>;
+
+		/// The overarching call type.
+		type Call: Parameter
+			+ Dispatchable<Origin = Self::Origin>
+			+ GetDispatchInfo
+			+ From<frame_system::Call<Self>>
+			+ From<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::Call>;
+
+		// Used to interface with the Oracle.
+		type ForeignStateQuery: ForeignChainStateQuery<Self::AccountId, <Self as Config>::Call>;
+
+		/// Interface used to communicate with the NFT module.
+		type NFTInterface: Inspect<Self::AccountId, ClassId = Self::ClassId, InstanceId = Self::TokenId>
+			+ Mutate<Self::AccountId>
+			+ InspectExtended<Self::AccountId>
+			+ Create<Self::AccountId>
+			+ CreateExtended<Self::AccountId, Properties, Balance = Balance>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The account's NFT token cannot be found.
+		AccountTokenNotFound,
+		/// The caller is unauthorized to make this transaction.
+		CallerUnauthorized,
+		/// The owner of the NFT has insufficient reserve balance.
+		InsufficientReservedBalance,
+		/// Cannot decode data from oracle
+		BadOracleData,
+		/// Failed to prove account spawned anonymous proxy
+		FailedAnonymousCheck,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The user has requested to mint a Account Token NFT.
+		MintRequested { account: T::AccountId, who: T::AccountId },
+		/// The mint request is deemed invalid by oracle.
+		MintRequestRejected { requester: T::AccountId },
+		/// A Account Token NFT is minted to an account.
+		AccountTokenMinted {
+			owner: T::AccountId,
+			account: T::AccountId,
+			token_id: T::TokenId,
+		},
+		/// An request to burn the account token is submitted. XCM message is sent to the
+		/// relaychain.
+		BurnRequested {
+			account: T::AccountId,
+			owner: T::AccountId,
+			token_id: T::TokenId,
+			new_owner: T::AccountId,
+		},
+		/// The account token is burned, the control of the `account` on the relaychain is
+		/// relinquished to `new_owner`.
+		AccountTokenBurned {
+			account: T::AccountId,
+			owner: T::AccountId,
+			token_id: T::TokenId,
+			new_owner: T::AccountId,
+		},
+	}
+
+	/// Stores the NFT's class ID. Created on RuntimeUpgrade. Used to mint and burn PRTs.
+	#[pallet::storage]
+	#[pallet::getter(fn nft_class_id)]
+	pub type NFTClassId<T: Config> = StorageValue<_, T::ClassId, ValueQuery>;
+
+	/// Stores proxy accounts that are already minted as an Account Token NFT.
+	/// Storage Map: Tokenized Account Id  => NFT Token ID
+	#[pallet::storage]
+	#[pallet::getter(fn minted_account)]
+	pub type MintedAccount<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, T::TokenId, OptionQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Create the NFT class once.
+		fn on_runtime_upgrade() -> Weight {
```

Comment:
need to make sure the storage version is initialized correctly for a new chain, and perform the create NFT task in build genesis spec so that those state are valid for any new chain.

Reply:
Added the genesis config and build